pkcs12/tests/kdf.rs

```diff
@@ -0,0 +1,130 @@
+use cfg_if::cfg_if;
+
+cfg_if! {
+    if #[cfg(feature="alloc")] {
+
+        #[test]
+        fn pkcs12_key_derive_sha256() {
```

Comment:
Can you provide some provenance information for these test vectors?

Reply:
I have basically written a small programming calling the openssl API to get the reference results. I will see if I find some more transparent way to document the validity of the tests.

Reply:
Can you check in the code you used to generate it? (provided it's Rust)

Otherwise it might be possible to generate such vectors using the `openssl` CLI, possibly using a shell script to generate them, which would be nice.

`openssl asn1parse` is quite useful for inspecting the contents of ASN.1-encoded documents.

Reply:
You can find the tool I used to generate the test cases here: https://github.com/xemwebe/test_pkcs12_kdf/tree/main/src .
However, recent versions of `openssl` does not seem to support the `whirlpool` algorithm by default any longer.
Unfortunately, I haven't found a way yet to generate these vectors with the `openssl` CLI.

Reply:
Okay, can you at least add a comment with a link to your tool?

Reply:
I have added a link to reference the test program.